homeassistant/components/switch/rest.py

```diff
@@ -36,26 +37,38 @@ def setup_platform(hass, config, add_devices_callback, discovery_info=None):
         _LOGGER.error("No route to resource/endpoint: %s", resource)
         return False
 
+    cfg_headers = config.get('headers')
+
+    if cfg_headers is None:
+        headers = DEFAULT_HEADERS
+    else:
+        headers = {}
+        for header in cfg_headers:
```

Comment:
Yes, but it seems to me they are a different, so I had to re-arrange it in order to make it work.
The actual Headers are sent; I've tested it and works as desired with the sample configuration.

My Python knowledge is limited, so I would love to see this PR accepted and code fixed (so I can learn how to do it properly). 

Thanks for your support!